include/AudioEngineProfiler.h

```diff
@@ -44,20 +45,37 @@ class AudioEngineProfiler
 		m_periodTimer.reset();
 	}
 
-	void finishPeriod( sample_rate_t sampleRate, fpp_t framesPerPeriod );
+	void finishPeriod(sample_rate_t sampleRate, fpp_t framesPerPeriod);
 
 	int cpuLoad() const
 	{
 		return m_cpuLoad;
 	}
 
-	void setOutputFile( const QString& outputFile );
+	void setOutputFile(const QString& outputFile);
 
+	enum DetailType {
```

Comment:
We could use ``enum class`` instead, in light of efforts made in f102777.

Reply:
Sure, why not. I did not even know `enum class` existed. :)

Reply:
Hm, can `enum class` be still used for array indexing etc.? Because after the change I'm getting:
> error: could not convert ‘lmms::AudioEngineProfiler::DetailType::DetailCount’ from ‘lmms::AudioEngineProfiler::DetailType’ to ‘long unsigned int’

which suggests that use-case is completely unsupported..

Reply:
> Hm, can `enum class` be still used for array indexing etc.? Because after the change I'm getting:
> 
> > error: could not convert ‘lmms::AudioEngineProfiler::DetailType::DetailCount’ from ‘lmms::AudioEngineProfiler::DetailType’ to ‘long unsigned int’
> 
> which suggests that use-case is completely unsupported..

``enum class`` does not allow implicit conversions to integral types. Try making the conversion explicit using ``static_cast``.